build/AzurePipelinesTemplates/WindowsAppSDK-BuildProject-Steps.yml

```diff
@@ -66,7 +66,7 @@ steps:
       vsVersion: 16.0
       platform: '$(buildPlatform)'
       configuration: '$(buildConfiguration)'
-      msbuildArgs: '/p:AppxSymbolPackageEnabled=false /binaryLogger:$(Build.SourcesDirectory)/${{ parameters.solutionPath }}.$(buildPlatform).$(buildConfiguration).binlog /p:WindowsAppSDKVersionBuild=$(builddate_yymm) /p:WindowsAppSDKVersionRevision=$(builddate_dd)$(buildrevision) /p:VCToolsInstallDir="$(VCToolsInstallDir)\" /p:PGOBuildMode=$(PGOBuildMode) /p:WindowsAppSDKBuildPipeline=1'
+      msbuildArgs: '/restore /p:AppxSymbolPackageEnabled=false /binaryLogger:$(Build.SourcesDirectory)/${{ parameters.solutionPath }}.$(buildPlatform).$(buildConfiguration).binlog /p:WindowsAppSDKVersionBuild=$(builddate_yymm) /p:WindowsAppSDKVersionRevision=$(builddate_dd)$(buildrevision) /p:VCToolsInstallDir="$(VCToolsInstallDir)\" /p:PGOBuildMode=$(PGOBuildMode) /p:WindowsAppSDKBuildPipeline=1'
```

Comment:
Please remove

`/restore` does what? Do a `NuGet restore`?

That's unnecessary. The build pipeline explicitly does that in build\AzurePipelinesTemplates\WindowsAppSDK-BuildProject-Steps.yml line 40+ (`...displayName: 'NuGet restore ${{ parameters.solutionPath }}'...`) plus some other prep work before this task

And split across 2 (or more - see build-mrt.yml) is a good thing. If something goes wrong with the NuGet operation it's easier to identify (vs here mixed in with the build logic)

Reply:
yes. /restore does a restore for PackageReference. "nuget restore" doesn't fully respect directory.build.props, and puts project.assets.json in wrong place. Also it's recommended to restore using msbuild.

Reply:
Please follow up with @BenJKuhn and @kythant as what you describe implies the current pipeline definition is defective and needs broader changes

As /restore is unrelated to moving to /objs I suggest removing the /restore change and addressing it in a separate PR

Reply:
not defective, but cannot handle the custom path. /restore is a related change. otherwise those projects importing nuget packages with PackageReference will fail to build.

Reply:
Then we should remove the explicit Restore tasks before this?

Why are we doing *both*?

Reply:
that's still needed for packages.config. Figuring out who they are and only restore those before build (that's actually what MRTCore does) is out of scope of this change.